app/controllers/recipe_foods_controller.rb

```diff
@@ -0,0 +1,51 @@
+class RecipeFoodsController < ApplicationController
+  def new
+    @recipe_food = RecipeFood.new
+    @current_user = current_user
+    @recipe = Recipe.find(params[:recipe_id])
+  end
+
+  def create
+    user = current_user
+    recipe = Recipe.find(params[:recipe_id])
+    recipe_food = RecipeFood.new({
+                                   quantity: recipe_food_params['qty'],
+                                   recipe_id: recipe.id,
+                                   food_id: recipe_food_params['food_id']
+                                 })
+    if recipe_food.save
+      flash[:success] = 'created successfully'
+    else
+      flash[:alert] = 'error creating recipe'
+    end
+    redirect_to user_recipe_path(user_id: user.id, id: recipe.id)
+  end
+
+  def edit
+    @current_user = current_user
+    @recipe = Recipe.find(params[:recipe_id])
+    @recipe_food = RecipeFood.find(params[:id])
+  end
+
+  def update
+    @recipe_item = RecipeFood.find(params[:id])
+    @recipe_item.update_column(:quantity, update_food_params['qty'])
+    redirect_to user_recipe_path(user_id: params[:user_id], id: params[:recipe_id])
+  end
+
+  def destroy
+    recipe_food = RecipeFood.find(params[:id])
+    recipe_food.destroy
+    redirect_to user_recipe_path(user_id: params[:user_id], id: params[:recipe_id])
+  end
```

Comment:
- [ ] It will be great to create success and fail paths for your destroy action. If the resource is deleted, do something. If it is not deleted do something else. This will prevent your application from misbehaving incase something unexpected happens

Reply:
@peterrobert not part of the project requirements. you should have marked this as optional.

Reply:
No, this is not optional, when you implement either create destroy or update in your application, is a requirement for us to check if you have done it right. And the only way to do it right is to ensure you have created both success and fail paths. @ram1117